### PR TITLE
lxc: Print "permission denied" when LXD socket isn't writable

### LINKED ISSUE #12761: Confusing message if `/var/snap/lxd/common/lxd/unix.socket` exists but isn't writable
Steps to reproduce:

```
$ lxc launch ubuntu-daily:22.04 v1 --vm
$ lxc shell v1
root@v1:~# snap refresh lxd --channel latest/edge
lxd (edge) git-675b31b from Canonical✓ refreshed

# Wake LXD up
root@v1:~# lxc ls

# Remove ubuntu from lxd group:
root@v1:~# deluser ubuntu lxd
Removing user `ubuntu' from group `lxd' ...
Done.

# Try to access LXD without being member of the right group:
root@v1:~# su - ubuntu
ubuntu@v1:~$ lxc ls
...
Error: The LXD daemon doesn't appear to be started (socket path: /var/snap/lxd/common/lxd/unix.socket)
```

Yet that socket is there and LXD is started:
```
ubuntu@v1:~$ ll /var/snap/lxd/common/lxd/unix.socket
srw-rw---- 1 root lxd 0 Jan 22 18:36 /var/snap/lxd/common/lxd/unix.socket=
ubuntu@v1:~$ systemctl status snap.lxd.daemon | cat
● snap.lxd.daemon.service - Service for snap application lxd.daemon
     Loaded: loaded (/etc/systemd/system/snap.lxd.daemon.service; static)
     Active: active (running) since Mon 2024-01-22 18:39:00 UTC; 2min 15s ago
TriggeredBy: ● snap.lxd.daemon.unix.socket
   Main PID: 1599 (daemon.start)
      Tasks: 0 (limit: 1169)
     Memory: 24.0M
        CPU: 448ms
     CGroup: /system.slice/snap.lxd.daemon.service
             ‣ 1599 /bin/sh /snap/lxd/26701/commands/daemon.start

Jan 22 18:39:01 v1 lxd.daemon[1754]: - proc_uptime
Jan 22 18:39:01 v1 lxd.daemon[1754]: - proc_slabinfo
Jan 22 18:39:01 v1 lxd.daemon[1754]: - shared_pidns
Jan 22 18:39:01 v1 lxd.daemon[1754]: - cpuview_daemon
Jan 22 18:39:01 v1 lxd.daemon[1754]: - loadavg_daemon
Jan 22 18:39:01 v1 lxd.daemon[1754]: - pidfds
Jan 22 18:39:02 v1 lxd.daemon[1599]: => Starting LXD
Jan 22 18:39:03 v1 lxd.daemon[1766]: time="2024-01-22T18:39:03Z" level=warning msg=" - Couldn't find the CGroup network priority controller, per-instance network priority will be ignored. Please use per-device limits.priority instead"
Jan 22 18:39:06 v1 lxd.daemon[1599]: => First LXD execution on this system
Jan 22 18:39:06 v1 lxd.daemon[1599]: => LXD is ready
```

LXD's socket is just not writable:
```
ubuntu@v1:~$ test -w /var/snap/lxd/common/lxd/unix.socket; echo $?
1
```

IMHO, `lxc` should give a better error message if the Unix socket is not writable.

### PR DESCRIPTION
**Pull request by MggMuggins**
Resolves #12761

### CODE REVIEW

File: lxc/config/remote.go [hunk 1]
```diff
@@ -78,12 +78,20 @@ func (c *Config) GetInstanceServer(name string) (lxd.InstanceServer, error) {
...
 
 			if errors.As(err, &netErr) {
-				return nil, fmt.Errorf("The LXD daemon doesn't appear to be started (socket path: %s)", netErr.Addr)
+				socketPath = fmt.Sprintf(" (socket path: %s)", netErr.Addr)
 			}
 
-			return nil, err
+			switch {
+			case errors.Is(err, os.ErrNotExist):
+				return nil, fmt.Errorf("The LXD daemon doesn't appear to be started%s", socketPath)
```
Comment (tomponline) [inline lxc/config/remote.go:89]: Suggest doing something a little simpler and more informative here:

```go
return nil, fmt.Errorf("LXD unix socket file %q not found. Please check LXD is running", netErr.Addr)
```

and 

```go
return nil, fmt.Errorf("LXD unix socket file %q not accessible . Please check the file permissions", netErr.Addr)
```

File: lxc/config/remote.go [hunk 2]
```diff
@@ -78,12 +78,20 @@ func (c *Config) GetInstanceServer(name string) (lxd.InstanceServer, error) {
...
+			case errors.Is(err, os.ErrPermission):
+				return nil, fmt.Errorf("Permission denied%s", socketPath)
+			default:
+				return nil, err
```
Comment (tomponline) [inline lxc/config/remote.go:93]: Also here lets add error wrapping as follows:

```go
return nil, fmt.Errorf("LXD unix socket file %q not accessible: %w", netErr.Addr, err)
```

File: lxc/config/remote.go (hunk 1 above)
Comment (MggMuggins) [reply lxc/config/remote.go]: I was shooting to reduce the line length a little, but I agree that this is a marked improvement.

File: lxc/config/remote.go (hunk 2 above)
Comment (MggMuggins) [reply lxc/config/remote.go]: While having the error be `Unwrap`able by the caller would be ideal, the message gets pretty gnarly:
```
Error: LXD unix socket "/var/lib/lxd/unix.socket" not accessible: Get "http://unix.socket/1.0": dial unix /var/lib/lxd/unix.socket: connect: permission denied
```
I'm happy to use `%w` if you'd like but I thought a more concise message is probably worth the slightly less useful `error`.

Comment (tomponline) [reply lxc/config/remote.go]: I think we should only do the error wrapping for the unknown error scenario - i.e not a permission denied or not found errors which will have the nicer error messages. But we should prefix it with an explanation as to what the general problem is.

**Comment by tomponline** [comment]
Thanks! Is this tested working OK in the different scenarios?

**Comment by MggMuggins** [comment]
Yes, I ran through each branch (and commented out the specific checks to run L91 just to be safe). I can add a test if you'd like.

**Review by tomponline — APPROVED** [review]

**Pull request merged**